src/config/field-overrides/program-rules/programRuleProgramStage.js

```diff
@@ -9,31 +10,46 @@ class ProgramStageField extends React.Component {
         //Need this in state, or else the dropdown will reload all the time, since the
         //filter is a new reference
         this.state = {
-            queryParamFilter: null,
-            programId: null,
+            queryParamFilter: this.getQueryParamFilter(),
+            programId: (props.model.program && props.model.program.id) || null,
         };
     }
 
+    getQueryParamFilter = (props = this.props) => {
+        if(!props.model.program) {
+            return null;
+        }
+        return [`program.id:eq:${props.model.program.id}`];
+    }
+
     componentWillReceiveProps(newProps) {
         const selectedProgram = newProps.model.program;
         if (selectedProgram && selectedProgram.id !== this.state.programId) {
-            const queryParamFilter = [`program.id:eq:${selectedProgram.id}`];
+            const queryParamFilter = this.getQueryParamFilter(newProps);
 
             this.setState({
                 programId: selectedProgram.id,
                 queryParamFilter
             });
+            //Clear programStage when program is changed
+            objectActions.update({fieldName: 'programStage', value: null})
         }
     }
 
-    render() {
+    shouldRender = () => {
         const props = this.props;
-        const disabled = !this.state.programId;
+        const isEventProgram = props.model.program && props.model.program.programType === "WITHOUT_REGISTRATION";
+        return props.model.program && !isEventProgram;
+    }
+
```

Comment:
I think the whole thing would be slightly cleaner like this:
```javascript 
shouldRender() {
    const { program } = this.props.model;
    return !!(program && program.programType === 'WITHOUT_REGISTRATION');
}
```

Reply:
Yeah, you're right, a mistake in my suggestion. I missed the `!` you had before `isEventProgram`.
It's definitely not worth changing this, since these were mostly cosmetic suggestions.